cpp/src/parquet/arrow/reader.cc

```diff
@@ -1060,7 +1077,10 @@ class RowGroupGenerator {
     }
     auto ready = reader->parquet_reader()->WhenBuffered({row_group}, column_indices);
     if (cpu_executor_) ready = cpu_executor_->TransferAlways(ready);
-    return ready.Then([=]() -> ::arrow::Future<RecordBatchGenerator> {
+
+    GET_CURRENT_SPAN(span);
```

Comment:
This is a confusing use of macros.  At the very least I'd want to see something like `GET_CURRENT_SPAN(&span)`

Reply:
I agree that would be better, but then the span object declaration is not guarded with the `#ifdef 
 ARROW_WITH_OPENTELEMETRY` (that was the reason I did it like this). But I'll try it and see if it works.